### PR TITLE
add show merges

### LINKED ISSUE #55652: `SHOW MERGES` query
**Use case**

Why not?

### PR DESCRIPTION
**Pull request by jetgm**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add support for SHOW MERGES query.

Closes https://github.com/ClickHouse/ClickHouse/issues/55652

### Documentation entry for user-facing changes

image

### CODE REVIEW

**Comment by den-crane** [comment]
BTW, I use another query:

```
SELECT
    database,
    table,
    round((elapsed * (1 / progress)) - elapsed, 2) AS estimate_complete,
    elapsed,
    round(progress, 2) AS progress,
    is_mutation,
    formatReadableSize(total_size_bytes_compressed) AS size,
    formatReadableSize(memory_usage) AS mem
FROM system.merges

┌─database─┬─table───────────────────┬─estimate_complete─┬──────elapsed─┬─progress─┬─is_mutation─┬─size───────┬─mem────────┐
│ dw       │ csp_ad_fact_event_shard │             28.29 │ 84.876010993 │     0.75 │           0 │ 1.38 GiB   │ 21.08 MiB  │
│ dw       │ csp_ad_fact_event_shard │             74.13 │ 43.539542221 │     0.37 │           0 │ 282.96 MiB │ 1.73 GiB   │
│ csplogs  │ log_row_u_shard         │             32.01 │   1.33372388 │     0.04 │           0 │ 529.64 MiB │ 241.41 MiB │
│ dw       │ log_row_csp_a_shard     │              2.25 │  0.093698493 │     0.04 │           0 │ 96.86 MiB  │ 79.96 MiB  │
└──────────┴─────────────────────────┴───────────────────┴──────────────┴──────────┴─────────────┴────────────┴────────────┘
```

**Comment by Algunenano** [comment]
2 conflicting PRs now -> https://github.com/ClickHouse/ClickHouse/pull/55770

**Comment by jetgm** [comment]
> round((elapsed * (1 / progress)) - elapsed, 2) AS estimate_complete,
>     elapsed,
>     round(progress, 2) AS progress,
>     is_mutation,
>     formatReadableSize(total_size_bytes_compressed) AS size,
>     formatReadableSize(memory_usage) AS mem

Your sql is better for this

image

**Comment by elenatorro** [comment]
Hey! I already opened [a PR](https://github.com/ClickHouse/ClickHouse/pull/55770) with this functionality, but since it's being discussed here how to improve the query to use, we can push this one forward 👍🏽

File: docs/en/sql-reference/statements/show.md [hunk 1]
```diff
@@ -651,3 +651,47 @@ If either `LIKE` or `ILIKE` clause is specified, the query returns a list of sys
...
+
+Query:
+
+``` sql
+SHOW MERGES;
+```
+
+Result:
+
+```text
```
Comment (Algunenano) [inline docs/en/sql-reference/statements/show.md:676]: The examples need to be updated with the output of the new query

File: src/Interpreters/InterpreterShowTablesQuery.cpp [hunk 2]
```diff
@@ -116,6 +116,32 @@ String InterpreterShowTablesQuery::getRewrittenQuery()
...
+        if (!query.like.empty())
+        {
+            rewritten_query
+                << " WHERE table "
+                << (query.not_like ? "NOT " : "")
+                << (query.case_insensitive_like ? "ILIKE " : "LIKE ")
+                << DB::quote << query.like;
+        }
+
+        /// (*)
```
Comment (Algunenano) [inline src/Interpreters/InterpreterShowTablesQuery.cpp:136]: Extra unnecessary comment

File: src/Interpreters/InterpreterShowTablesQuery.cpp [hunk 3]
```diff
@@ -116,6 +116,32 @@ String InterpreterShowTablesQuery::getRewrittenQuery()
...
+        rewritten_query << " ORDER BY elapsed desc, database, table";
```
Comment (Algunenano) [inline src/Interpreters/InterpreterShowTablesQuery.cpp:137]: Do we really need `database, table` as part of the query? Isn't `elapsed` enough?

**Review by Algunenano — CHANGES_REQUESTED** [review]

File: src/Interpreters/InterpreterShowTablesQuery.cpp (hunk 2 above)
Comment (Algunenano) [reply src/Interpreters/InterpreterShowTablesQuery.cpp:136]: Nevermind, I see there was already a comment with the `(*)` explaining what they mean

File: docs/en/sql-reference/statements/show.md (hunk 1 above)
Comment (jetgm) [reply docs/en/sql-reference/statements/show.md:676]: Updated.

File: src/Interpreters/InterpreterShowTablesQuery.cpp (hunk 3 above)
Comment (jetgm) [reply src/Interpreters/InterpreterShowTablesQuery.cpp]: Have changed, only elapsed remained.

**Comment by Algunenano** [comment]
Failures in `Integration tests (asan, analyzer) [1/6]`:
* `test_zookeeper_config/test.py`: Marked as allowed to fail in the analyzer, but they didn't fail so it gets messy.
* `test_timezone_config/test.py::*`. Docker timed out when setting up the CH instances. Not related.

**Review by Algunenano — APPROVED** [review]
Nice. Thank you!

**Pull request merged**